app/src/main/java/deakin/gopher/guardian/view/nextofkin/NoKAddFragment.java

```diff
@@ -175,12 +175,8 @@ private void saveNextofKin() {
     if (null == nextofKin) {
       nextofKin = new NextOfKin(firstName, middleName, lastName, address, phoneNumber, email, null);
     } else {
-      nextofKin.setFirstName(firstName);
-      nextofKin.setMiddleName(middleName);
-      nextofKin.setLastName(lastName);
-      nextofKin.setHomeAddress(address);
-      nextofKin.setMobilePhone(phoneNumber);
-      nextofKin.setEmailAddress(email);
+      nextofKin = new NextOfKin(firstName,middleName,lastName,address,phoneNumber,email,null);
```

Comment:
Constructing a new `NextOfKin` here isn't right, and this has the exact same logic as the `if` part of this conditional. You should do the `setter` calls like how it was previously.